addons/event/models/event_event.py

```diff
@@ -494,13 +505,19 @@ def _compute_ticket_instructions(self):
                is_html_empty(event.event_type_id.ticket_instructions):
                 event.ticket_instructions = event.event_type_id.ticket_instructions
 
-    @api.constrains('seats_max', 'seats_available', 'seats_limited')
-    def _check_seats_limit(self):
+    @api.constrains('seats_max', 'seats_limited', 'registration_ids')
+    def _check_seats_availability(self, n_registrations=0):
```

Comment:
A bit strange usage of the constraint ... why this argument ? Shouldn't we simply have a constraint based on registration_ids.state and use seats_available or directly do a search_count on registrations when a registration is created or when its state is updated ?